plugins/ipstack/icon_ipstack/connection/connection.py

```diff
@@ -13,4 +16,19 @@ def connect(self, params):
         self.token = params.get(Input.CRED_TOKEN).get("secretKey")
 
     def test(self):
-        return {"success": True}
+        # Sending a request using the secretKey to test if it is valid
+        url = "http://api.ipstack.com/" + "rapid7.com" + "?access_key=" + self.token + "&output=json"
+
+        # These contain custom status codes (101, 102, 104, 106) returned from IPStack, as seen in the lookup action.py
+        invalid_status_codes = ["404", "101", "102", "104", "106"]
```

Comment:
The reason we have to use status codes in this way is due to IPStack always returning a 200, even when failing. This is the output for a failed test :

"rapid7/IPStack:3.0.2\nresponse.msg= 'OK'\nresponse.getcode()= 200\nparsed_response= {'success': False, 'error': {'code': 101, 'type': 'missing_access_key', 'info': 'You have not supplied an API Access Key. [Required format: access_key=YOUR_ACCESS_KEY]'}}\n".

I have changed it to check the `success` key as it differs when the test is valid or not.